main.go

```diff
@@ -1,9 +1,441 @@
-// Copyright 2017 The hubble-fit Authors.  All rights reserved.
+// Copyright 2017 The hubble-fit Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
 // hubble-fit is a program to fit the Hubble diagram with a Cosmological model.
 package main
 
+import (
+	"fmt"
+	"image/color"
+	"io/ioutil"
+	"log"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/gonum/integrate"
+	"github.com/gonum/optimize"
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/plotter"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+	"github.com/gonum/stat"
+	"go-hep.org/x/hep/fit"
+)
+
+/*
+#####
+Constants
+#####
+*/
+
+const (
+	c       = 299792458	// speed of light
+	H       = 0.070		// Hubble constant
+	omgM    = 0.295		// Mass density parameter, = (8*pi*G*rho_mass)/(3*H²)
+	alpha   = 0.141		// free parameter for the Hubble fit (factor for stretch)
+	beta    = 3.101		// free parameter for the Hubble fit (factor for color)
+	Mb      = -19.05	// absolute blue magnitude of a 1A SN
+	delta_M = -0.070	// uncertainty on the absolute blue magnitude
+)
+
+/*
+#####
+Basic analysis functions
+#####
+*/
+
+func Sx(s []float64) float64 {
+
+	// compute the standard deviation of a given distribution
+
+	var sum, sigma float64
+	for _, i := range s {
+		sum += i * i
+	}
+
+	sigma = math.Sqrt(sum / (float64(len(s)) - 1))
+
+	return sigma
+}
+
+func RMS(s []float64) float64 {
+
+	// compute the root mean square of a given distribution
+
+	var sum, rms float64
+	for _, i := range s {
+		sum = +i * i
+	}
+
+	rms = math.Sqrt(sum / float64(len(s)))
+
+	return rms
+}
+
+func RMSerr(s []float64) float64 {
+
+	// compute the error on the root mean square of a given distribution
+
+	var rmserr float64
+	rmserr = Sx(s) / math.Sqrt(2*float64(len(s)))
+
+	return rmserr
+}
+
+func MEANerr(s []float64) float64 {
+
+	// compute the error on the mean of a given distribution
+
+	var meanerr float64
+	meanerr = Sx(s) / math.Sqrt(float64(len(s)))
+
+	return meanerr
+}
+
+func difference(s1, s2 []float64) ([]float64, []float64) {
+
+	// compute the difference between two slices, value by value
+
+	diff := make([]float64, len(s1))
+	abs := make([]float64, len(s1))
+
+	for i := range(s1) {
+		val := s2[i] - s1[i]
+		diff[i] = val
+		abs[i] = math.Abs(val)
+	}
+
+	return diff, abs
+}
+
+func AtoF (s string) float64 {
+
+	// convert a string to float64
+
+	v, err := strconv.ParseFloat(s, 64)
+	if err != nil {
+		log.Panic(err)
+	}
+	return v
+}
+
+/*
+#####
+Theoretical mu
+#####
+*/
+
+func integral(z, n float64) float64 {
+
+	// compute the integral of the D_l function, based on the trapeze method
+	// integrate.Trapezoidal takes two slices (abscissa and ordinate) from which to compute the integral
+	// inputs : z = redshift, n = number of steps for the trapeze method
+
+	abs := make([]float64, int(n+1))
+	ord := make([]float64, int(n+1))
+
+	for i := 0; i <= int(n); i++ {
```

Comment:
these 2 loops could be combined into 1:

```go
invN := 1 / n
for i := range abs {
    x := z * float64(i) * invN
    y := 1 / math.Sqrt((1+x*omgM)*(1+x)(1+x)-(1-omgM)*(2+x)*x
    abs[i] = x
    ord[i] = y
}
```

also, better rename `abs` and `ord` as `xs` and `ys`.